### PR TITLE
Add Docker-ized basic stack for React, Next, and TypeScript (#14)

### LINKED ISSUE #14: Getting the Initial Local setup going with React Stack
This issue can be characterized having 
1. Have a Docker setup
2. Integrating with React Ecosystem along with ExpressJS. Use of NextJS is the Developer choice if faster for development.
3. Integrating with LTI launch, this can be addressed as separate issue. Database of choice seems to be Postgres but confirmation from MIDatabase  https://teamdynamix.umich.edu/TDNext/Apps/31/Tickets/TicketDet.aspx?TicketID=633962 would help back our decision.

### PR DESCRIPTION
**Pull request by ssciolla**
Compare to #24. Features consistent in both include the following:
* docker-compose setup for development
* live reloading on save across the stack
* Sample API route

The PR resolves #14.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,18 @@
-## Canvas Course Manager
\ No newline at end of file
+## Canvas Course Manager
```
Comment (pushyamig) [inline README.md:1]: This seems straight forward, lot of complexities are lifted by NEXT.JS framework it self.

**Comment by chrisrrowland** [comment]
docker-compose up fails for me
`sh: 1: next: not found`

however
```
npm run build
npm run dev 
```
works

**Comment by ssciolla** [comment]
> docker-compose up fails for me

Thanks, I will see if I can reproduce, but not planning to spend a lot of time on this since we're going down the Express path.

**Comment by chrisrrowland** [comment]
Yeah I wouldn't bother unless you're just really curious.

**Comment by ssciolla** [comment]
@chrisrrowland, I can't seem to reproduce this problem. What steps did you follow to set this up? You can tell me offline, whenever you have time.

**Comment by ssciolla** [comment]
I'm closing this PR, as we went with #24 instead.

**Pull request closed without merging**